site/gatsby-site/src/components/IncidentReportCard.js

```diff
@@ -0,0 +1,297 @@
+export default function IncidentReportCard(props) {
```

Comment:
What's the reason for using the same component to render an incident and a report?

My idea always has been to have `/components/reports/ReportCard` and `/components/incidents/IncidentCard`, (considering that rendering of each might diverge in the future)

Reply:
There's more divergence in how we render reports in one place and the next than in how we render reports vs incidents. For instance, the list-mode discover cards representing reports to me look more like the horizontal incident card on the landing page than the the report cards on the citation page. For that reason, this component is very flexible in its rendering, but while providing a unified visual design. It can have a vertical or horizontal layout. It can have children as headers or footers. It can render different subsets of the data. And it can render data that represents a report or an incident – or something else. This PR uses it for the candidates, and it could probably be used for submissions too if we someday wanted that. Most of its present and future capabilities are independent of what data it's rendering. Let's say that we make it so that hovering over the date shows a tooltip with a more precise date. Well, that's something we'd probably want for both incidents and reports. Most foreseeable changes are like that. If something is applicable only to one or the other, like showing the taxonomy classifications, it seems likely to be better implemented as a child component than as an addition to the card component.

Reply:
Sorry I meditated a little bit on this. I get what you say; the line between incidents and reports is a bit blurry, but too many conditionals in the render function are a code smell; we are forcing an abstraction that isn't there. If you had to unit test this component, you'd end up with tons of tests, which makes me think this is not the best approach. 

For practical purposes, as we develop new features, it is better to have some separation (even at the cost of duplication/inconsistencies) when you have many people working on the same project. I get flashbacks from past projects with things breaking here and there because the same component is used everywhere :shudders:

Reply:
I've thought about this too, and you're right. I think the thing to do is separate it into different components based on the display logic rather than incidents vs reports vs candidates vs issues vs... Additionally, separate out child components for things like the subtitle that can be shared between the different display types.